src/pages/staticPages/tools/oncoprinter/OncoprinterToolUtils.ts

```diff
@@ -21,3 +26,21 @@ export function getDataForSubmission(
         }
     });
 }
+
+export function getDefaultClinicalAttributeColoringForStringDatatype(
+    data: ClinicalTrackDatum[]
+) {
+    const colorGetter = makeUniqueColorGetter(
+        _.values(RESERVED_CLINICAL_VALUE_COLORS)
+    );
+    let categoryToColor = _.cloneDeep(RESERVED_CLINICAL_VALUE_COLORS);
```

Comment:
i think you can avoid deep cloning here by instead merging RESERVED_CLINICAL_VALUE_COLORS onto a map made from data.  This will be clearer also.   That way RESERVED_CLINICAL_VALUE_COLORS will always win.